src/components/app/details/triggerView/BranchRegexModal.tsx

```diff
@@ -0,0 +1,160 @@
+                if (_ciMaterial.gitMaterialId === gitMaterialId && _ciMaterial.source) {
+                    if (_ciMaterial.source?.type === SourceTypeMap.BranchRegex) {
+                        return _ciMaterial.source.regex
+                    }
+                }
+            }
+        }
+        return ''
+    }
+
+    const renderBranchRegexMaterialHeader = (close) => {
+        return (
+            <div className="trigger-modal__header">
+                <h1 className="modal__title flex left fs-16">{title}</h1>
+                <button
+                    type="button"
+                    className="transparent"
+                    onClick={() => {
+                        close()
+                    }}
+                >
+                    <Close className="" />
+                </button>
+            </div>
+        )
+    }
+
+    const renderMaterialRegexFooterNextButton = (context) => {
+        return (
+            <div className="trigger-modal__trigger flex right">
+                <button
+                    className="cta flex"
+                    onClick={(e) => {
+                        onClickNextButton(context)
+                    }}
+                >
+                    Save & Next
+                    <LeftIcon style={{ ['--rotateBy' as any]: '180deg' }} className="rotate icon-dim-16 ml-8 scn-0 " />
+                </button>
+            </div>
+        )
+    }
+
+    const renderValidationErrorLabel = (message?: string): JSX.Element => {
+        return (
+            <div className="error-label flex left align-start fs-11 fw-4 mt-6 ml-36">
+                <Error className="icon-dim-16" />
+                <div className="ml-4 cr-5">{message}</div>
+            </div>
+        )
+    }
+
+    const onClickBackArrow = () => {
+        onCloseBranchRegexModal()
+        onShowCIModal()
+    }
+
+    return (
+        <div>
+            <>
+                <div>{renderBranchRegexMaterialHeader(context.closeCIModal)}</div>
+
+                <div className="select-material--regex-body m-20 fs-13">
+                    <div className="flex left">
+                        {isChangeBranchClicked && (
+                            <div onClick={onClickBackArrow}>
+                                <LeftIcon className="rotate icon-dim-20 mr-16 cursor" />
+                            </div>
+                        )}
+
+                        <div>
+                            <h4 className="mb-0 fw-6 ">Set a primary branch</h4>
+                            <p className="mt-4">
+                                Primary branch will be used to trigger automatic builds on every commit. This can be
+                                changed later.
+                            </p>
+                        </div>
+                    </div>
+
+                    {material &&
+                        material.map((mat, index) => {
+                            const _regexValue = regexValue[mat.gitMaterialId]
```

Comment:
Can we provide some default value in _regexValue so that we don't need to check for undefined multiple times?